File: src/Block/BeginWhileBlock.java
/**
 * This file copyright (c) 2015-2016, Jing Fan, Adalbert Gerald Soosai Raj, and Jignesh M. Patel
 *
 * See the file CREDITS.txt in the root directory for details.
 **/

package Block;

/**
 * @brief Begin while block.
 *
 */
public class BeginWhileBlock extends Block {

  /**
   * @brief Constructor.
   * @param stage The string indicates the stage of this code block.
   * @param indent The indent level.
   * @param endStr The string indicates the termination condition of the
   *               iterations.
   */
  public BeginWhileBlock(String stage, int indent, String endStr) {
    super(stage, indent);
    String initVal = "-1";
    if (!endStr.equals("NO_MESSAGE")) {
      initVal = endStr.substring(endStr.indexOf('(') + 1,
                                 endStr.indexOf(')'))
                      .split(",")[1].trim();
    }
    this.append("DECLARE @flag int");
    this.append("SET @flag = " + initVal);
    this.append("WHILE @flag != 0");
    this.append("BEGIN");
    this.sql = this.sb.toString();
  }

}

File: python/src/Blocks/BeginWhileBlock.py
from Block import Block

class BeginWhileBlock(Block):
    '''
    Generates sql statements for a begin while block
    '''
    def __init__(self,stage,indent, loopCondition):
        '''
        constructor
        @param stage: current stage  
        @type stage: string 
        @param indent: indent level
        @type indent: int
        @param loopCondition: condition inside the while loop
        @type loopCondition: string
        '''
        super(BeginWhileBlock,self).__init__(stage,indent)
        self.append("BEGIN"+"\n")
        self.append("WHILE "+loopCondition+" LOOP")
        self.sql=self.sb 